python/tvm/driver/tvmc/autotuner.py

```diff
@@ -0,0 +1,301 @@
+    #     can be improved in future to add integration with a modelzoo
+    #     or URL, for example.
+    parser.add_argument("FILE", help="path to the input model file")
+
+
+def drive_tune(args):
+    """Invoke auto-tuning with command line arguments
+
+    Parameters
+    ----------
+    args: argparse.Namespace
+        Arguments from command line parser.
+
+    Returns
+    --------
+    int
+        Zero if successfully completed
+
+    """
+
+    target = common.target_from_cli(args.target)
+    mod, params = frontends.load_model(args.FILE, args.model_format)
+
+    tasks = get_tuning_tasks(
+        mod=mod,
+        params=params,
+        target=target,
+        alter_layout=args.desired_layout,
+    )
+
+    if args.hostname:
+        logging.info("starting remote tuning:")
+        logging.info(" hostname: %s", args.hostname)
+        logging.info(" port    : %s", args.port)
+        if not args.tracker_key:
+            raise common.TVMCException(
+                "need to provide tracker key (--tracker-key) for remote tuning"
+            )
+
+        runner = autotvm.RPCRunner(
+            key=args.tracker_key,
+            host=args.hostname,
+            port=args.port,
+            number=args.number,
+            repeat=args.repeat,
+            n_parallel=args.parallel,
+            timeout=args.timeout,
+            min_repeat_ms=args.min_repeat_ms,
+        )
+    else:
+        logging.info("starting localhost tuning")
+        runner = autotvm.LocalRunner(
+            number=args.number,
+            repeat=args.repeat,
+            timeout=args.timeout,
+            min_repeat_ms=args.min_repeat_ms,
+        )
+
+    tuning_option = {
+        "tuner": args.tuner,
+        "trials": args.trials,
+        "early_stopping": args.early_stopping,
+        "measure_option": autotvm.measure_option(
+            builder=autotvm.LocalBuilder(build_func="default"), runner=runner
+        ),
+        "tuning_records": args.tuning_records,
+    }
+    logging.debug(" tuning options: %s", tuning_option)
+
+    tune_tasks(tasks, args.output, **tuning_option)
+    return 0
+
+
+def get_tuning_tasks(mod, params, target, target_host=None, alter_layout=None):
+    """Get the tuning tasks for a given relay module.
+
+    Parameters
+    ----------
+    mod : tvm.relay.Module
+        The relay module from which to extract tuning tasks.
+    params : dict
+        The params for the relay module.
+    target : tvm.target.Target
+        The compilation target.
+    target_host : str, optional
+        The compilation target for the host.
+    alter_layout : str, optional
+        The layout to convert the graph to. Note, the convert layout
+        pass doesn't currently guarantee the whole of the graph will
+        be converted to the chosen layout.
+
+    Returns
+    -------
+    tasks : list of autotvm.Tasks
+        list of tasks to be tuned
+
+    """
+    if not target_host:
+        target_host = target
```

Comment:
Is this correct? What if `target` is `cuda`?

Reply:
looking into this

Reply:
Based on the existing tutorials that I had a look, it seems sensible just to completely remove `target_host` from here (and below, line 244), rather than expose it via some `--target-host TARGET_HOST`. What do you think?

Reply:
Let's expose it via `--target-host` with `llvm` as its default value.

Reply:
Understood. Done now.